kolibri/plugins/learn/assets/src/views/TopicsPage/index.vue

```diff
@@ -620,56 +617,58 @@
       channelTitle() {
         return this.channel ? this.channel.name : '';
       },
-      resources() {
-        return this.contents.filter(content => content.kind !== ContentNodeKinds.TOPIC);
-      },
       childrenToDisplay() {
         return this.windowBreakpoint === 2 || this.windowBreakpoint > 4 ? 4 : 3;
       },
       gridType() {
         return this.windowBreakpoint > 4 ? 2 : 1;
       },
-      resourcesDisplayed() {
-        // if no folders are shown at this level, show more resources to fill the space
-        // or if the user has explicitly requested to show more resources
-        if (!this.topics.length || this.showMoreResources) {
-          return this.resources;
-        }
-        return this.resources.slice(0, this.childrenToDisplay);
-      },
-      moreResources() {
-        return this.resourcesDisplayed.length < this.resources.length;
-      },
       topics() {
         return this.contents
           .filter(content => content.kind === ContentNodeKinds.TOPIC)
-          .filter(t => t.children && t.children.results.length)
-          .map(t => {
-            let topicChildren = t.children.results;
+          .filter(t => t.children && t.children.results.length);
+      },
+      contentsForDisplay() {
+        return this.contents
+          .filter(
+            c =>
+              c.kind !== ContentNodeKinds.TOPIC ||
+              ((this.subTopicId ? c.id === this.subTopicId : true) &&
+                c.children &&
+                c.children.results.length),
+          )
+          .reduce((arr, c) => {
+            // Reduce the list to objects representing topics,
+            // and arrays representing runs of resources positioned between them.
```

Comment:
So, some of this understanding I'm pulling in from the issue/design work, but this is for a situation where at the same level within a topic tree, there is a combination of folders and resources: 

├── Chemistry
│   ├── syllabus.txt
│   ├── lesson 1
│   │   ├── chemical_bonds.mp4
│   ├── readme.txt

we want to create a nested array that has groups of either topics or arrays of single resources, with a preserved order?

Reply:
So, for the above example, the output array is of the shape:
```
[
 [ { syllabus.txt }],
 { lesson 1 },
 [ { readme.txt }],
]
```

So, resources that are adjacent to each other are grouped together in an array to be passed to the component for displaying a grid of resources, while the topics are left as top level objects to be displayed by the SubTopic component.

Reply:
okay great - that was my understanding. good to be sure that I'm following